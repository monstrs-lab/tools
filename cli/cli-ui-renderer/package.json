{
  "name": "@monstrs/cli-ui-renderer",
  "version": "0.3.3",
  "license": "MIT",
  "type": "module",
  "exports": {
    "./package.json": "./package.json",
    ".": "./src/index.ts"
  },
  "main": "src/index.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "yarn library build",
    "prepack": "yarn run build",
    "postpack": "rm -rf dist"
  },
  "dependencies": {
    "ink": "^4.2.0",
    "react": "^18.2.0"
  },
  "devDependencies": {
    "@types/node": "^20.3.0",
    "@types/react": "^18.2.13"
  },
  "publishConfig": {
    "access": "public",
    "exports": {
      "./package.json": "./package.json",
      ".": {
        "import": "./dist/index.js",
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      }
    },
    "main": "dist/index.js",
    "types": "dist/index.d.ts"
  }
}
