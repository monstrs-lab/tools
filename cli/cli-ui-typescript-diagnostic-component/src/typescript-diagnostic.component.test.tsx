import React                    from 'react'
import stripAnsi                from 'strip-ansi'

import { renderStatic }         from '@monstrs/cli-ui-renderer'

import { TypeScriptDiagnostic } from './typescript-diagnostic.component'

describe('eslint result component', () => {
  it('render', () => {
    const value = {
      file: {
        fileName:
          '/Users/andreylinko/workspace/lab/tools/cli/cli-ui-typescript-diagnostic-component/src/b.ts',
        text: "const n = (s: number) => s\nn('asdfasdfasd')\n",
        lineMap: [0, 27, 44],
      },
      start: 29,
      category: 1,
      messageText: "Argument of type 'string' is not assignable to parameter of type 'number'.",
    }

    const output = renderStatic(<TypeScriptDiagnostic {...value} />, 160)

    expect(stripAnsi(output)).toMatchSnapshot()
  })
})
