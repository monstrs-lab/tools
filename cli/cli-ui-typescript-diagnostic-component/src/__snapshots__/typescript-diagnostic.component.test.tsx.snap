// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslint result component render 1`] = `
"cli/cli-ui-typescript-diagnostic-component/src/b.ts:2:2

  Error: Argument of type 'string' is not assignable to parameter of type 'number'.

  1 | const n = (s: number) => s
> 2 | n('asdfasdfasd')
    |  ^
  3 |

"
`;
